presto-main/src/main/java/com/facebook/presto/operator/WorkProcessor.java

```diff
@@ -127,6 +127,33 @@
 
     interface Transformation<T, R>
     {
+        /**
+         * Processes input elements and returns current transformation state.
+         *
+         * @param elementOptional an element to be transformed. Will be empty
+         * when there are no more elements. In such case transformation should
+         * finish processing and flush any remaining data.
+         * @return a current transformation state.
+         * <ul>
+         * <li>if state is {@link ProcessorState#finished()} then transformation has finished
+         * and {@link Transformation#process(java.util.Optional)} won't be called again;</li>
```

Comment:
nit: `s/{@link Transformation#process(java.util.Optional)}/this method/` or `the process method` (here & below)

nit: no need for `;` at the end of list items